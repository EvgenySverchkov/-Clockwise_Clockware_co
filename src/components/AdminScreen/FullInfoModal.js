import React from "react";
import PropTypes from "prop-types";

function FullInfoModal({ itemObj }) {
  return (
    <div
      className="modal fade"
      id="fullInfoModal"
      tabIndex="-1"
      role="dialog"
      aria-labelledby="exampleModalLongTitle"
      aria-hidden="true"
    >
      <div className="modal-dialog" role="document">
        <div className="modal-content">
          <div className="modal-header">
            <h5 className="modal-title" id="exampleModalLongTitle">
              Full info
            </h5>
            <button
              type="button"
              className="close"
              data-dismiss="modal"
              aria-label="Close"
            >
              <span aria-hidden="true">&times;</span>
            </button>
          </div>
          <div className="modal-body">
            {Object.keys(itemObj).map((key) => {
              return (
                <div key={key}>
                  {key}: <span>{itemObj[key]}</span>
                </div>
              );
            })}
          </div>
        </div>
      </div>
    </div>
  );
}

export default FullInfoModal;
