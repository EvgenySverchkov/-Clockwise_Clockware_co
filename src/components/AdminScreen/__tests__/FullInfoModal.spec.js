import React from 'react';
import {create} from "react-test-renderer";

import FullInfoModal from "../FullInfoModal";

it("Test of <FullInfoModal />", ()=>{
    const mockProps = {
        itemObj:{
            id: 1,
            rating: 5,
            name: "Petrovich",
            towns: ""
        }
    }
    const component = create(<FullInfoModal {...mockProps}/>);
    expect(component.toJSON()).toMatchSnapshot();
});