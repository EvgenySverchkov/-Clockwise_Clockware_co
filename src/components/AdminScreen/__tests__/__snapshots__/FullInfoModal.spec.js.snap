// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <FullInfoModal /> 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="exampleModalLongTitle"
  className="modal fade"
  id="fullInfoModal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="exampleModalLongTitle"
        >
          Full info
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <div>
          id
          : 
          <span>
            1
          </span>
        </div>
        <div>
          rating
          : 
          <span>
            5
          </span>
        </div>
        <div>
          name
          : 
          <span>
            Petrovich
          </span>
        </div>
        <div>
          towns
          : 
          <span>
            
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
